app/models/category.rb

```diff
@@ -207,7 +207,7 @@ def ensure_slug
 
     if slug.present?
       # santized custom slug
-      self.slug = Slug.for(slug)
+      self.slug = Slug.safe_for(slug)
       errors.add(:slug, 'is already in use') if duplicate_slug?
     else
       # auto slug
```

Comment:
I don't think so. As the topic described, custom slug should still be available in that circumstance since this site setting is used for maximizing the ability to customized non-english slug.